src/main/java/io/vertx/core/http/HttpServer.java

```diff
@@ -218,4 +218,10 @@ public interface HttpServer extends Measured {
    * @return the actual port the server is listening on.
    */
   int actualPort();
+
+  /**
+   * Attempts to reload the ssl configuration if the server has been configured with ssl
+   */
+  void reloadSsl();
```

Comment:
that should be an asynchronous operation that might fail

Reply:
Let me refactor it to an async call. But you assume it will fail, how?

Reply:
And... I am not sure how to make this operation async, can you share an example maybe? I found out I could use a Future, Promise, AsyncResult but I am having trouble understanding the best practice with it

Reply:
I attempted to make the reloading of ssl material async and it works, next to that the tests also passes 😄 Can you have another look at the latest changes?